### PR TITLE
Replacing rimraf in target-dir-clean with non-node.js based solution

### PR DESCRIPTION
**Pull request by moozzyk**
rimraf needs to pull node.js and is slow.

### CODE REVIEW

File: build/_k-standard-goals.shade [hunk 1]
```diff
@@ -65,7 +65,19 @@ default Configuration='${E("Configuration")}'
   k-restore
 
 #target-dir-clean target='clean'
-  rimraf rimrafDir='${TARGET_DIR}' if='Directory.Exists(TARGET_DIR)'
+  @{
+      if(Directory.Exists(TARGET_DIR))
```
Comment (dougbu) [inline build/_k-standard-goals.shade]: space after `if`

File: build/_k-standard-goals.shade [hunk 2]
```diff
@@ -65,7 +65,19 @@ default Configuration='${E("Configuration")}'
...
+      {
+          var directory = new DirectoryInfo(TARGET_DIR) { Attributes = FileAttributes.Normal };
+
+          foreach (var info in directory.GetFileSystemInfos("*", SearchOption.AllDirectories))
+          {
+              info.Attributes = FileAttributes.Normal;
```
Comment (dougbu) [inline build/_k-standard-goals.shade]: This loses all other attributes set in `info`.  Use `|=` instead and also do same update separately for `directory`.

**Comment by dougbu** [comment]
:watch:

File: build/_k-standard-goals.shade (hunk 2 above)
Comment (moozzyk) [inline build/_k-standard-goals.shade]: Why do we care if we are deleting the files?

Comment (dougbu) [inline build/_k-standard-goals.shade]: The system won't let you change some files to `Normal` but would let you remove them. So my `|=` suggestion should have been `&= ~FileAttributes.ReadOnly`. `ReadOnly` is the only bit worth twiddling here.

Comment (moozzyk) [inline build/_k-standard-goals.shade]: I see. Will do and update the PR.

**Comment by moozzyk** [comment]
@dougbu updated the PR as per your comments

**Comment by dougbu** [comment]
:shipit:

**Pull request merged**